src/Core/AdminConsole/Services/Implementations/RabbitMqIntegrationListenerService.cs

```diff
@@ -0,0 +1,180 @@
+        await _channel.QueueBindAsync(queue: _retryQueueName,
+                                      exchange: _exchangeName,
+                                      routingKey: _retryRoutingKey,
+                                      cancellationToken: cancellationToken);
+
+        // Declare dead letter queue
+        await _channel.QueueDeclareAsync(queue: _deadLetterQueueName,
+                                         durable: true,
+                                         exclusive: false,
+                                         autoDelete: false,
+                                         arguments: null,
+                                         cancellationToken: cancellationToken);
+        await _channel.QueueBindAsync(queue: _deadLetterQueueName,
+                                      exchange: _exchangeName,
+                                      routingKey: _deadLetterRoutingKey,
+                                      cancellationToken: cancellationToken);
+
+        await base.StartAsync(cancellationToken);
+    }
+
+    protected override async Task ExecuteAsync(CancellationToken cancellationToken)
+    {
+        var consumer = new AsyncEventingBasicConsumer(_channel);
+        consumer.ReceivedAsync += async (_, ea) =>
+        {
+            var json = Encoding.UTF8.GetString(ea.Body.Span);
+
+            try
+            {
+                var result = await _handler.HandleAsync(json);
+                var message = result.Message;
+
+                if (result.Success)
+                {
+                    await _channel.BasicAckAsync(ea.DeliveryTag, false, cancellationToken);
+                    return;
+                }
+
+                if (result.Retryable)
+                {
+                    message.ApplyRetry(result.NotBeforeUtc);
+
+                    if (message.RetryCount < _maxRetries)
+                    {
+                        await _channel.BasicPublishAsync(
+                            exchange: _exchangeName,
+                            routingKey: _retryRoutingKey,
+                            body: Encoding.UTF8.GetBytes(message.ToJson()),
+                            cancellationToken: cancellationToken);
+                    }
+                    else
+                    {
+                        await PublishToDeadLetterAsync(message.ToJson());
+                        _logger.LogWarning("Max retry attempts reached. Sent to DLQ.");
+                    }
+                }
+                else
+                {
+                    await PublishToDeadLetterAsync(message.ToJson());
+                    _logger.LogWarning("Non-retryable failure. Sent to DLQ.");
+                }
+
+                await _channel.BasicAckAsync(ea.DeliveryTag, false, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Unhandled error processing integration message");
```

Comment:
```suggestion
                _logger.LogError(ex, "Unhandled error processing integration message.");
```

⛏️ Maybe? Let's just be consistent.